barman/cli.py

```diff
@@ -918,47 +930,66 @@ def recover(args):
         )
         output.close_and_exit()
 
-    # If the backup to be recovered is compressed or an incremental then
-    # there are additional checks to be carried out
-    if backup_id.compression is not None or backup_id.parent_backup_id is not None:
-        # Set the recovery staging path from the cli if it is set
-        if args.recovery_staging_path is not None:
-            try:
-                recovery_staging_path = parse_recovery_staging_path(
-                    args.recovery_staging_path
-                )
-            except ValueError as exc:
-                output.error("Cannot parse recovery staging path: %s", str(exc))
-                output.close_and_exit()
-            server.config.recovery_staging_path = recovery_staging_path
+    # If the user sets both flags, we output an error and exit.
+    # This prevents from staging_path being overwritten in the
+    # code "block" below.
+    if args.local_staging_path is not None and args.recovery_staging_path is not None:
+        output.error(
+            "Cannot set `recovery_staging_path` and "
+            "`local_staging_path` to recover.",
+        )
+        output.close_and_exit()
```

Comment:
I think we should refactor this whole "staging path" validation here.

It might be clearer if we simply use two different `if` blocks, one for handling compression/recovery_staging_path, another for `incremental/local_staging_path`.

We used to have this code in Barman 3.10.1 for handling the former case: https://github.com/EnterpriseDB/barman/blob/release/3.10.1/barman/cli.py#L898-L925.
I think we should have that back, and add a similar block for the latter case.